raster-foundry/raster-foundry
# Lab UI - Implement reclassify modal histogram

## Overview

This PR implements the histogram within the reclassify settings modal. It also changes other related components to make the reclassification UI (node, table, etc) functional.

### Checklist

~- [ ] Styleguide updated, if necessary~
~- [ ] Swagger specification updated, if necessary~
~- [ ] Symlinks from new migrations present or corrected for any new migrations~

### Demo

![image](https://user-images.githubusercontent.com/2442245/30927856-d0e18ad6-a387-11e7-91c3-46700a38eff4.png)

![image](https://user-images.githubusercontent.com/2442245/30927912-00fb7722-a388-11e7-95f5-8fdb9cd2a2d8.png)

![image](https://user-images.githubusercontent.com/2442245/30927953-192c3a2a-a388-11e7-8048-633319f04205.png)

### Notes

The wireframes indicate a hover action on the histogram where hovering over the histogram between two breaks will display the value it is mapped to. This is not implemented due to time constraints.

This makes some changes to the histogram breakpoint component, which is used in other place. Obviously, the intent here was to preserve the existing functionality so there are some tests below for that.

## Testing Instructions

 * Open a tool that uses reclassify
 * Try adjusting the reclassification values and make sure they stick when exiting the modal
 * Check that the resulting node histogram reflects the reclassification
 * Preview the map on that node

 * Test the breakpoint dragging and interactions on the color scheme/masking histograms to make sure I didn't break anything

Closes #2447
